src/main/java/io/kestra/plugin/gcp/pubsub/model/Message.java

```diff
@@ -35,10 +37,11 @@ public class Message {
     @PluginProperty(dynamic = true)
     private String orderingKey;
 
-    public PubsubMessage to(RunContext runContext) throws IllegalVariableEvaluationException {
+    public PubsubMessage to(RunContext runContext, SerdeType serdeType) throws IllegalVariableEvaluationException, IOException {
         var builder =  PubsubMessage.newBuilder();
         if(data != null) {
-            builder.setData(ByteString.copyFrom(runContext.render(data).getBytes()));
+            var serializedData = serdeType.serialize(data);
```

Comment:
Data isn't rendered anymore?

Reply:
Seems like a bug :(

Reply:
Fixed, can you have a look?

Reply:
Sry to be annoying, but can you add the case if people provide a Map<String, Object> and render with [this](https://github.com/kestra-io/kestra/blob/1694df7c26dd66897c76769b479e8382a1864f7b/core/src/main/java/io/kestra/core/runners/RunContext.java#L432-L432), so people can provide object that contains task outputs

Reply:
SQS didn't have this functionality neither, this is a different thing.